src/gtc/cuir/cuir_codegen.py

```diff
@@ -16,14 +16,35 @@
 
 from typing import Any, Collection, Dict, List, Set, Union
 
-from eve import codegen, traits
+from eve import NodeVisitor, codegen, traits
 from eve.codegen import FormatTemplate as as_fmt
 from eve.codegen import MakoTemplate as as_mako
 from eve.concepts import LeafNode
 from gtc.common import BuiltInLiteral, DataType, LevelMarker, NativeFunction, UnaryOperator
 from gtc.cuir import cuir
 
 
+def check_int(s: str) -> bool:
```

Comment:
Maybe prepend and underscore to make it “private”. Or replace by a local function as proposed below.